Beta_Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py

```diff
@@ -0,0 +1,354 @@
+        self.token: str = ''
+        self.expiry_time: int = 0
+
+    async def set_access_token(self, client: Client) -> None:
+        self.client = client
+        if not self.token:
+            token = await self._get_access_token()
+            self.token = token
+        client.set_auth_headers(self.token)
+
+    async def _get_access_token(self) -> str:
+        """Retrieves CrowdStrike Falcon access token required for API requests.
+
+        Runs for max retries of 3 attempts in case of API rate limit hit.
+
+        Returns:
+            str: The access token retrieved.
+
+        Raises:
+            RuntimeError: An error occurred (json.decoder.JSONDecodeError) trying to deserialize the API response.
+        """
+        token = None
+        body = None
+        max_retries = 3
+        async with ClientSession(
+            connector=TCPConnector(ssl=self.verify_ssl),
+            headers=TOKEN_RETRIEVAL_HEADERS,
+            trust_env=self.proxy
+        ) as session:
+            for _ in range(max_retries):
+                data = {
+                    'client_id': self.client_id,
+                    'client_secret': self.client_secret
+                }
+                async with session.post(f'{self.base_url}/oauth2/token', data=data) as res:
+                    if res.status == TOO_MANY_REQUESTS_STATUS_CODE:
+                        demisto.debug('Token retrieval requests status: rate limit exceeded, will retry in 5 seconds.')
+                        await sleep(5)
+                    elif res.status == CREATED_STATUS_CODE:
+                        try:
+                            body = await res.json()
+                            break
+                        except json.decoder.JSONDecodeError:
+                            raise RuntimeError(
+                                f'Failed to decode successful token retrieval response: {str(res.content)}'
+                            )
+                    else:
+                        body = await res.json()
+                        try:
+                            body = await res.json()
+                            error = body.get('errors', [{}])
+                            error_message = error[0].get('message', '')
+                            raise RuntimeError(
+                                f'Failed to retrieve token, verify client details are correct: {error_message}'
+                            )
+                        except json.decoder.JSONDecodeError:
+                            raise RuntimeError(
+                                f'Failed to decode token retrieval failure response: {str(res.content)}'
+                            )
+            if not body:
+                raise RuntimeError('Failed to retrieve token')
+            token = body.get('access_token')
+            self.expiry_time = body.get('expires_in', MINUTES_30) - TIME_BUFFER_1_MINUTE
+        if not token:
+            raise RuntimeError('Failed to retrieve token')
```

Comment:
same error message as in line 237, might be hard to to debug from logs in case of error